sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -315,7 +322,80 @@ public override long Position
 
         public override long Seek(long offset, SeekOrigin origin)
         {
-            throw new NotSupportedException();
+            long newPosition = CalculateNewPosition(offset, origin);
+
+            if (newPosition == _position)
+            {
+                return _position;
+            }
+
+            // newPosition < 0
+            if (newPosition < 0)
+            {
+                throw new ArgumentException($"New {nameof(offset)} cannot be less than 0.  Value was {newPosition}");
+            }
+
+            // newPosition > _length
+            if (newPosition > _length)
+            {
+                if (_allowBlobModifications)
+                {
+                    _position = newPosition;
+                    _seeked = true;
+                    return newPosition;
+                }
+                else
+                {
+                    throw new ArgumentException(
+                        $"{nameof(newPosition)} exceeds known blob or file length.  This condition is not allowed with allowBlobModifications == false.");
```

Comment:
`allowBlobModifications` doesn't make much sense when you're working with a generic `System.IO.Stream` if this error gets raised randomly in the guts of some utility code.  Maybe we could say something like `You cannot seek past the end of a stream created by OpenReadStream that does not allow modifications to the underlying blob or file.`

Reply:
fixed.